extraction/embedded_tweet_extractor.py

```diff
@@ -0,0 +1,22 @@
+"""This script checks whether DOM has embedded tweet tag or not and
+creates and returns the ETweet object"""
+
+from extraction.data_models.embedded_tweet import ETweet
+from extraction.interface_content_extractor import IContentExtractor
+
+
+class ETweetExtractor(IContentExtractor):
+    """This class inherits IContentExtractor for extracting embedded tweets"""
+
+    def validate_and_extract(self, node):
+        try:
+            if node['class'] == ['twitter-tweet', 'twitter-tweet-rendered'] \
+                    or node['class'] == ['twitter-tweet']:
+                tweet_a_tag = node.find_all('a')[-1]
```

Comment:
`tweet_url` instead of `tweet_a_tag`

Reply:
It will return '<a>' tag, that's why the name.

Reply:
I meant, once you do ['href']

Reply:
Updated accordingly!